packages/api/src/controllers/asset.ts

```diff
@@ -200,6 +202,13 @@ export async function validateAssetPayload(
     }
   }
 
+  if (payload.projectId) {
+    const project = await getProject(req);
```

Comment:
I wonder if we shouldn't add `project` to `req` in https://github.com/livepeer/studio/blob/95e12f9b0136a60ee0ff5c55608c1f5bd7be3278/packages/api/src/middleware/auth.ts#L79

My thinking is that now, `authenticator()` assigns `user` to each request. But from now on, the request (api token) will be associated always with one project, so maybe we should inject it in `authenticator()` and then there would be no need to call `getProject()` everywhere?

Reply:
@victorges and I discussed something like that but Victor suggested we need to maintain compatibility with the SDK I believe. I'm still a bit on the fence on this b/c from the dashboard perspective, project is basically a "filter" to categorize the assets, streams, etc. But maybe philosophically what you're saying makes sense. wdyt @victorges?

Reply:
I think we can still maintain compatibility but having the projectId added to the request as Rafal suggests will just neaten the code up a bit. User and project would both exist on the request object then, and we can maintain compatibility by checking both in the controllers.

Reply:
The other thing is we don't want projectId to be part of the payload, we should be taking it from the api key being used. We could potentially leave that for a later PR though and leave this one mainly as just adding the projects controller.

Reply:
@mjh1: I saw your edit comment too late and started making simliar changes to my branch - but I scrapped those and just cherry-picked yours here instead. Could we use this branch to pair for integration?

Reply:
Guys, I think you're on the right track!

Wrt what @mjh1 wrote, I think that we should not send `projectId` in the payload, because `projectId` is derived from API TOKEN. The same way, we don't send `userId` in every request.

Reply:
Ok cool yep let's just use this branch